File: day1/excercise 2/c#/HotelRoom.cs
public class HotelRoom
{
    private bool _isAvailable = true;

    public HotelRoom(int roomNumber, decimal pricePerNight)
    {
        ArgumentOutOfRangeException.ThrowIfNegativeOrZero(roomNumber);
        ArgumentOutOfRangeException.ThrowIfNegativeOrZero(pricePerNight);

        RoomNumber = roomNumber;
        PricePerNight = pricePerNight;
    }

    public int RoomNumber { get; }
    public decimal PricePerNight { get; }
    public bool IsAvailable => _isAvailable;

    public void Book()
    {
        if (!_isAvailable)
            throw new InvalidOperationException("Room is already booked.");

        _isAvailable = false;
    }

    public void Checkout()
    {
        if (_isAvailable)
            throw new InvalidOperationException("Room is already available.");

        _isAvailable = true;
    }
}

File: day1/excercise 2/node-express/hotelRoom.js
class HotelRoom {
    #available = true;

    constructor(roomNumber, pricePerNight) {
        if (roomNumber <= 0) {
            throw new Error("Room number must be greater than zero.");
        }

        if (pricePerNight <= 0) {
            throw new Error("Price must be greater than zero.");
        }

        this.roomNumber = roomNumber;
        this.pricePerNight = pricePerNight;
    }

    get isAvailable() {
        return this.#available;
    }

    book() {
        if (!this.#available) {
            throw new Error("Room is already booked.");
        }

        this.#available = false;
    }

    checkout() {
        if (this.#available) {
            throw new Error("Room is already available.");
        }

        this.#available = true;
    }
}